vitals/Main.java

```diff
@@ -1,23 +1,64 @@
 package vitals;
 
 public class Main {
-    static boolean batteryIsOk(float temperature, float soc, float chargeRate) {
-        if(temperature < 0 || temperature > 45) {
-            System.out.println("Temperature is out of range!");
-            return false;
-        } else if(soc < 20 || soc > 80) {
-            System.out.println("State of Charge is out of range!");
-            return false;
-        } else if(chargeRate > 0.8) {
-            System.out.println("Charge Rate is out of range!");
-            return false;
-        }
-        return true;
+    
+  public static boolean isTemperatureOk(float temp) {
+    if(temp<0) {
+      System.out.println("Temperature is too low");
+      return false;
     }
+    else if(temp>45) {
+      System.out.println("Temperature is too high");
+      return false;
+    }
+    System.out.println("Temperature is OK");
+    return  true;
+  }
+  
+  public static boolean isChargingStateOk(float soc) {
+    if(soc<20) {
+      System.out.println("charging state is at very low for now, Dont remove charging");
+      return false;
+    }
+    else if(soc>80) {
+      System.out.println("charging state is at High, we can remove charging");
+      return false;
+    }
+    System.out.println("Vehicle is not charged yet, Keep charging ON");
+    return  true;
+  }
+  
+  
+  public static boolean isChargingRateOk(float chargeRate) {
+    if(chargeRate> 0.8) {
```

Comment:
usage of hard coded values will lead to issues during maintenance. if it gets used in many places and then a change in one place needs to be consistent with all others. There is a scope for improvement here.